direct.go

```diff
@@ -311,78 +316,88 @@ func cloneRequestWith(req *http.Request, frontedHost string, body io.ReadCloser)
 // tell us whether the masquerade is good or not (i.e. if masquerade was good,
 // keep it, else vet a new one).
 func (d *direct) dial() (net.Conn, *masquerade, func(bool) bool, error) {
-	conn, m, masqueradeGood, _, err := d.dialWith(d.masquerades)
+	conn, m, masqueradeGood, _, err := d.dialWith(d.masquerades, d.cached)
 	return conn, m, masqueradeGood, err
 }
 
-func (d *direct) dialWith(in chan masquerade) (net.Conn, *masquerade, func(bool) bool, bool, error) {
+func (d *direct) dialWith(vetted chan masquerade, cached chan masquerade) (net.Conn, *masquerade, func(bool) bool, bool, error) {
 	retryLater := make([]masquerade, 0)
 	defer func() {
+		log.Debugf("%d masquerades to retry", len(retryLater))
+		log.Debugf("capacity %d", len(vetted))
 		for _, m := range retryLater {
 			// when network just recovered from offline, retryLater has more
 			// elements than the capacity of the channel.
 			select {
-			case in <- m:
+			case vetted <- m:
 			default:
 				log.Debug("Dropping masquerade: retry channel full")
 			}
 		}
 	}()
 
 	for {
+		// order of preference vetted -> cached -> unvetted
```

Comment:
So what we do now is we always prefer vetted masquerades over cached ones, but we prefer cached over unvetted masquerades.